bsmetadata/metadata_utils.py

```diff
@@ -137,9 +175,11 @@ def create_global_metadata_prefix(example: Dict[str, Any], cfg: MetadataConfig)
         processor = PROCESSORS.get(key, MetadataProcessor)(cfg)
         processed_metadata[key] = processor.process_global(metadata)
 
-    sorted_metadata = [processed_metadata.get(md, None) for md in cfg.metadata_list]
-    sorted_metadata = [md for md in sorted_metadata if md is not None]
-    return cfg.metadata_sep.join(sorted_metadata) + cfg.global_metadata_sep if sorted_metadata else ""
```

Comment:
I changed the behavior of this function by not adding the separator between the global metadata string and the text string (`cfg.metadata_global_sep`) in the `create_global_metadata_prefix` method. This way, both types of separators (`cfg.metadata_global_sep` and `metadata_special_token_for_generation_sep`) are added within the `add_metadata_and_chunk_examples` method.

Reply:
outdated